pkg/controller/dual-pods/server-requesting.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2025 The llm-d Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dualpods
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"text/template"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+
+	"github.com/llm-d-incubation/llm-d-fast-model-actuation/pkg/api"
+	stubapi "github.com/llm-d-incubation/llm-d-fast-model-actuation/pkg/stub/api"
+)
+
+func (ctl *controller) processServerRequestingPod(ctx context.Context, requestingPod *corev1.Pod) (error, bool) {
+	logger := klog.FromContext(ctx)
+	logger.V(5).Info("Processing server-requesting pod", "name", requestingPod.Name)
+
+	serverPatch := requestingPod.Annotations[api.ServerPatchAnnotationName]
+	if serverPatch == "" {
+		logger.V(5).Info("No server patch annotation found", "name", requestingPod.Name)
```

Comment:
This is OK for a WIP, but the controller ought to react to erroneous user input by posting an error message in the Pod's status.